src/libraries/System.Private.CoreLib/src/System/Reflection/NullabilityInfoContext.cs

```diff
@@ -407,34 +397,19 @@ private NullabilityInfo GetNullabilityInfo(MemberInfo memberInfo, Type type, ILi
             return nullability;
         }
 
-        private static bool ParseNullableState(IList<CustomAttributeData> customAttributes, int index, ref NullabilityState state)
+        private static NullableAttributeStateParser CreateParser(IList<CustomAttributeData> customAttributes)
```

Comment:
By abstracting away the attributes behind the parser struct, it makes it a bit cleaner to support the public-only case. Also lets us avoid repeated searches of the attribute list which feels like a plus.